gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/TrinoQueueLengthChecker.java

```diff
@@ -48,18 +49,8 @@ public void observe(List<ClusterStats> stats)
                 continue;
             }
             if (!clusterQueueMap.containsKey(stat.getRoutingGroup())) {
-                clusterQueueMap.put(stat.getRoutingGroup(), new HashMap<String, Integer>()
-                        {
-                            {
-                                put(stat.getClusterId(), stat.getQueuedQueryCount());
-                            }
-                        });
-                clusterRunningMap.put(stat.getRoutingGroup(), new HashMap<String, Integer>()
-                        {
-                            {
-                                put(stat.getClusterId(), stat.getRunningQueryCount());
-                            }
-                        });
+                clusterQueueMap.put(stat.getRoutingGroup(), ImmutableMap.of(stat.getClusterId(), stat.getQueuedQueryCount()));
+                clusterRunningMap.put(stat.getRoutingGroup(), ImmutableMap.of(stat.getClusterId(), stat.getRunningQueryCount()));
```

Comment:
Is this change really safe? I can't find the existing test coverage. `else` block calls `ImmutableMap.put` which isn't supported.

Is it possible to use Table class in Guava?

Reply:
Good catch - will update

Reply:
Clarification - are you suggesting replacing the `Map<String, Map<String, Int>` with a `Table<String, String, Int>`? 

It doesn't look like there is a test case that covers the `else`. Maybe `TestClusterStatsMonitor` could be extended to test this

Reply:
Right - https://github.com/google/guava/wiki/CollectionUtilitiesExplained#tables